controllers/awscluster_controller.go

```diff
@@ -413,3 +428,123 @@ func (r *AWSClusterReconciler) requeueAWSClusterForUnpausedCluster(ctx context.C
 		}
 	}
 }
+
+func (r *AWSClusterReconciler) deleteAWSClusterFromIdentityOwnerRef(ctx context.Context, clusterScope *scope.ClusterScope) error {
+	identityObjectKey := client.ObjectKey{Name: clusterScope.IdentityRef().Name}
+	hasOwnerRef := false
+	switch clusterScope.IdentityRef().Kind {
```

Comment:
We don't need to know the kind of the identity here, just need to reach to ownerReferences. Unstructured struct could be used here.

Reply:
@Ankitasw I think you missed this comment. Is this doable?

Reply:
@sedefsavas totally forgot about this, will do it, thanks

Reply:
@sedefsavas it's ready now(still need to add unit tests which might take time)

Reply:
@sedefsavas I have added a scenario in unit test to check if finalizer and owner ref is updated properly. Do you think we need to cover any other scenario, not sure how to test in case of reconcileDelete() as after deleting cluster we can't check the identity ref?